guide_drafts/model-derives.md

```diff
@@ -0,0 +1,926 @@
+Model Derives in Depth
+===============
+
+Diesel lets you write safe queries by using traits that you can implement on your models.
+Writing these implementations by hand can be very laborious, so in addition to the `diesel` crate, the `diesel_codegen` crate offers custom derives.
+In this section, we will cover each trait in detail in terms of its use cases, usage considerations, as well compare hand written implementations to its derive.
+To make use of these derives, `diesel_codegen` must be imported into the root of your project.
+
+Throughout this guide, we will be looking at examples for each trait and how they interact together.
+Some of the example code will be implementing basic [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) database operations.
+We'll be covering creating, reading, and updating data.
+The details of those operations will be not be covered beyond their relevance to the demonstrated trait.
+
+- [Queryable](#queryable)
+- [Insertable](#insertable)
+- [Identifiable](#identifiable)
+- [AsChangeset](#aschangeset)
+- [Associations](#associations)
+
+## <a name="queryable"></a>Queryable
+
+Using the [`#[derive(Queryable)]`](http://docs.diesel.rs/diesel/query_source/trait.Queryable.html)` trait annotation on your model struct allows records to be queried from the database.
+This means that any time you are using the `load()`, `get_result()`, `get_results()`, and `first()` methods to execute your queries, you must have `Queryable` implemented on your model. 
+`Queryable` doesn't directly provide these methods, but they are a "side effect" of the implementation.
+
+There are some considerations to be made when implementing the `Queryable` trait. 
+You must include all the table's fields in your model struct.
+The implementation details for this rule are only important if you wish to `impl` the trait by hand and will be covered later in this section.
+Your struct name should be the singular of its corresponding table name.
+Queryable by itself does not support the `#[table_name=""]` attribute. 
```

Comment:
Maybe combine these four sentences with conjunctions? They read like separate yet similar lists of facts.

Reply:
I originally thought this should be a bulleted list for a quick reference, but that is not how the rest of the guide is structured. 

How about...?
>  There are some considerations to be made when implementing the `Queryable` trait. First, you must include all the table's fields in your model struct. `Queryable` needs to know how many columns are in each row in order to `build()` that row. Also, your struct should be named in the singular form of its corresponding table name, as `Queryable` does not support the `#[table_name=""]` attribute.